### DISCUSSION TITLE
Why doesn't JsonNode have a ValueKind property?

### DISCUSSION THREAD

**Discussion by habbes** [dotnet/runtime#90917] [Q&A]
`System.Text.Json.JsonElement` strict as a `ValueKind` property that returns a `JsonValueKind` enum which allows us to query whether the value is a number, string, null, etc. `Newtonsoft.Json`'s `JToken` also has a similar `Type` property that returns a `JTokenType` enum. However, `System.Text.Json.Nodes.JsonNode` lacks such a property. Is there a reason that such an API is missing?

**Comment by Pyritie** [comment]
https://github.com/dotnet/runtime/issues/53406#issuecomment-890203807
